core/src/test/java/org/elasticsearch/indices/flush/SyncedFlushUtil.java

```diff
@@ -41,19 +45,23 @@ private SyncedFlushUtil() {
     /**
      * Blocking single index version of {@link SyncedFlushService#attemptSyncedFlush(String[], IndicesOptions, ActionListener)}
      */
-    public static IndicesSyncedFlushResult attemptSyncedFlush(InternalTestCluster cluster, String index) {
-        SyncedFlushService service = cluster.getInstance(SyncedFlushService.class);
-        LatchedListener<IndicesSyncedFlushResult> listener = new LatchedListener();
-        service.attemptSyncedFlush(new String[]{index}, IndicesOptions.lenientExpandOpen(), listener);
-        try {
-            listener.latch.await();
-        } catch (InterruptedException e) {
-            Thread.currentThread().interrupt();
+    public static SyncedFlushResponse attemptSyncedFlush(InternalTestCluster cluster, String index) {
+        if (randomBoolean()) {
+            SyncedFlushService service = cluster.getInstance(SyncedFlushService.class);
+            LatchedListener<SyncedFlushResponse> listener = new LatchedListener();
+            service.attemptSyncedFlush(new String[]{index}, IndicesOptions.lenientExpandOpen(), listener);
+            try {
+                listener.latch.await();
+            } catch (InterruptedException e) {
+                Thread.currentThread().interrupt();
+            }
+            if (listener.error != null) {
+                throw ExceptionsHelper.convertToElastic(listener.error);
+            }
+            return listener.result;
+        } else {
+            return client().admin().indices().prepareSyncedFlush(index).get();
```

Comment:
Is there a reason to do it the old way? I remember having trouble with one of the BWC tests because it wanted to synced flush but it couldn't because there was no API for it. If you _always_ do it with the client then the BWC test could actually use this method and just skip versions of ES before whatever this gets merged into.